### ISSUE TITLE
Zeus Integration

### ISSUE THREAD

**Issue by johncantrell97** [L2-Technology/sensei#21]
It should be pretty straight forward to implement a Zeus connecter for the child nodes.  I think most if not all of the api endpoints needed probably exist.  There might be some nice-to-have ones we can add as part of this.  Hardest part of this is probably setting up Zeus development environment to do testing, should probably talk with Evan about it.

**Comment by Bosch-0** [comment]
Would love to do this!

Are the channels that the child nodes have with the Sensei hosted channels for the added privacy?

**Comment by johncantrell97** [comment]
well, right now the nodes can open real channels to any node on the network (including the root node). I guess in theory we could add hosted channels as an option to avoid the fees if running in custodial mode (i guess?).  I haven't thought a lot about that yet but seems reasonable.

@louneskmt has started on this and surfaced these missing endpoints:

1) no decode payment request endpoint?
2) not possible to specify fee rate when closing a channel
3) there is a sign message endpoint but not a verify one
4) zero amount invoice are not supported
5) cannot set or get an invoice expiry date 
6) paid fees aren't returned for a paid invoice 

should be pretty easy to add all of these

**Comment by louneskmt** [comment]
Yep, my Sensei integration into Zeus is almost done (al least for all the basic stuff), I need to setup a Zeus dev env to test it out. 

@johncantrell97 has sent above the few points I've noticed while doing this 👌

**Comment by Bosch-0** [comment]
> I guess in theory we could add hosted channels as an option to avoid the fees if running in custodial mode (i guess?).

I could imagine on a very local / community level this may be okay. For example, a family uses sensei with HC so each member gets privacy / their own keys but they have shared custody.

Thanks for pointing these out we will get on to them asap. We have a big release coming up for the end of next month that will deal with most of these issues @kaloudis 

Zero amount invoices are in the latest release, what version are you using?

**Comment by louneskmt** [comment]
@Bosch-0 I think there was a misunderstanding: the exposed points concern Sensei, not Zeus. I was talking about things available in Zeus that aren't in Sensei endpoints (e.g. for zero amount invoices, the Sensei pay invoice endpoint only accepts an `invoice` string parameter, and not an `amount`).

**Comment by Bosch-0** [comment]
My bad, I misread. Though two of those points weren't in Zeus - made issues for them.

**Comment by louneskmt** [comment]
Another missing thing: channel fees (`base_fee_msat` and `fee_rate`). We can either return this directly in the channel details (`/v1/node/channels`) or have a dedicated endpoint @johncantrell97.
